addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -186,6 +189,29 @@ MockServer.include({
         });
         return messages;
     },
+    /**
+     * Simulate the 'message_post' Python method
+     *
+     * @private
+     * @return {integer}
+     */
+    _mockMessagePost(args) {
+        const {
+            args: [res_id],
+            model: res_model,
+            kwargs: postData,
+        } = args;
+        const model = this.data['mail.message'];
+        const messageIds = this.data['mail.message'].records.map(message => message.id);
+        const id = messageIds.length > 0 ? Math.max(...messageIds, 1) + 1 : 1;
```

Comment:
just `const id = Math.max(...messageIds, 0) + 1;`?

Reply:
I was not sure map will always return an empty array, will change that

Reply:
it will, you can't use `.map()` if records is not an array, and `map()` produce an array from another array and keeps same amount of items, hence `messageIds` here is always an array.